http/base/common/common.go

```diff
@@ -501,3 +504,191 @@ func GetAddress(str string) (common.Address, error) {
 	}
 	return address, err
 }
+
+type TxStateInfo struct {
+	TxID           string           // cross shard tx id: userTxHash+notify1+notify2...
+	Shards         map[uint64]uint8 // shards in this shard transaction, not include notification
+	NumNotifies    uint32
+	ShardNotifies  []XShardNotify
+	NextReqID      uint32
+	InReqResp      map[uint64][]XShardTxReqResp // todo: request id may conflict
+	PendingInReq   XShardTxReq
+	TotalInReq     uint32
+	OutReqResp     []XShardTxReqResp
+	TxPayload      string
+	PendingOutReq  XShardTxReq
+	PendingPrepare *xshard_types.XShardPrepareMsg
+	ExecState      uint8
+	Result         string
+	ResultErr      string
+	LockedAddress  []string
+	LockedKeys     []string
+	WriteSet       *overlaydb.MemDB
+	Notify         ExecuteNotify
+}
+type ShardMsgHeader struct {
+	ShardTxID     string
+	SourceShardID uint64
+	TargetShardID uint64
+	SourceTxHash  string
+}
+type XShardTxReq struct {
+	ShardMsgHeader
+	IdxInTx  uint64
+	Contract string
+	Payer    string
+	Fee      uint64
+	GasPrice uint64
+	Method   string
+	Args     string
+}
+type XShardTxRsp struct {
+	ShardMsgHeader
+	IdxInTx uint64
+	FeeUsed uint64
+	Error   bool
+	Result  string
+}
+type XShardTxReqResp struct {
+	Req   XShardTxReq
+	Resp  XShardTxRsp
+	Index uint32
+}
+
+type XShardNotify struct {
+	ShardMsgHeader
+	NotifyID uint32
+	Contract string
+	Payer    string
+	Fee      uint64
+	Method   string
+	Args     string
+}
+
+func ParseShardState(txState *xshard_state.TxState) (TxStateInfo, error) {
+
+	shards := make(map[uint64]uint8)
+	for k, v := range txState.Shards {
+		shards[k.ToUint64()] = uint8(v)
+	}
+	lockedAddress := make([]string, 0)
+	for _, addr := range txState.LockedAddress {
+		lockedAddress = append(lockedAddress, addr.ToBase58())
```

Comment:
addr.ToBase58() -> addr.ToHex(), because lockedAddress is contract address, use hex string is our habit.